plugins/module_utils/net_map/networking_definition.py

```diff
@@ -129,6 +129,36 @@ def _validate_parse_field_type(
     return raw_value
 
 
+def _validate_parse_trunk_parent_field(
+    field_name: str,
+    raw_definition: typing.Dict[str, typing.Any],
+    parent_name: str = None,
+    parent_type: str = None,
+    trunk_parents: set = set(),
```

Comment:
This is a Python trap. `set`, a mutable object, when used as default will make that every next calls to the function with no `trunk_parent` passed will default to the previous set instance. With other words, default mutable values acts as "globals". https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

Reply:
Good catch.
I changed it to default None, and then initialize a set if it is None.